### ISSUE TITLE
Second instrumentation module breaks whole maven build

### ISSUE THREAD

**Issue by outlying** [square/spoon#82]
We are using Maven and we have big project that contains 2 applications and common libraries. Recently I decided to write instrumentation tests for both application and launch them with Spoon. At second instrumentation test whole build failed

```[ERROR] Failed to execute goal com.squareup.spoon:spoon-maven-plugin:1.0.3:run (default) on project caregiver-it: Execution default of goal com.squareup.spoon:spoon-maven-plugin:1.0.3:run failed. NullPointerException -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.squareup.spoon:spoon-maven-plugin:1.0.3:run (default) on project caregiver-it: Execution default of goal com.squareup.spoon:spoon-maven-plugin:1.0.3:run failed.
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:225)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:84)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:59)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.singleThreadedBuild(LifecycleStarter.java:183)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:161)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:319)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:156)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:537)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:196)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:141)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:601)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:290)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:230)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:409)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:352)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default of goal com.squareup.spoon:spoon-maven-plugin:1.0.3:run failed.
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:110)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:209)
	... 19 more
Caused by: java.lang.NullPointerException
	at com.android.ddmlib.AndroidDebugBridge.stop(AndroidDebugBridge.java:691)
	at com.android.ddmlib.AndroidDebugBridge.createBridge(AndroidDebugBridge.java:296)
	at com.squareup.spoon.SpoonUtils.initAdb(SpoonUtils.java:121)
	at com.squareup.spoon.SpoonRunner.run(SpoonRunner.java:68)
	at com.squareup.spoon.SpoonMojo.execute(SpoonMojo.java:154)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:101)
	... 20 more```

Separately those tests work fine

**Comment by outlying** [comment]
Long time, I kinda forget about this issue and run into it again. I decided to create simple project that presents the issue:

https://github.com/outlying/secondspoon

Running
> ./gradlew clean spoon

Will end with error:
> AndroidDebugBridge.init() has already been called.

I used external Gradle plugin for Spoon this time but I think main issue is still the same. I will try to fix it in some spare time.

**Comment by edenman** [comment]
@outlying thanks for putting the test project together!  I just tried to repro with your test project and I didn't see the `AndroidDebugBridge.init() has already been called message`.  That said, I did see some issues with adb connections not working, and presumably that's related.  I've submitted #245 which I _think_ fixes the problem, but I'm not sure because I wasn't seeing the same symptom you were.

If you want to confirm (this is a pain): mvn install the branch in #245, then point spoon-gradle-plugin's build.gradle at spoon-runner:1.1.5-SNAPSHOT, gradle install it, then point your test project at mavenLocal() and spoon-gradle-plugin:0.14.2-SNAPSHOT.

Gonna close this out bc I think it's fixed: re-open if it's not.
